jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunction.kt

```diff
@@ -6,26 +6,53 @@ package software.aws.toolkits.jetbrains.services.lambda.upload
 import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.actionSystem.LangDataKeys
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
 import icons.AwsIcons
 import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationTemplateIndex
 import software.aws.toolkits.jetbrains.services.iam.IamRole
+import software.aws.toolkits.jetbrains.services.lambda.LambdaHandlerResolver
 import software.aws.toolkits.jetbrains.services.lambda.runtime
 import software.aws.toolkits.jetbrains.services.lambda.upload.EditFunctionMode.NEW
 import software.aws.toolkits.resources.message
 
-class CreateLambdaFunction(private val handlerName: String? = null) : AnAction(message("lambda.create_new"), null, AwsIcons.Actions.LAMBDA_FUNCTION_NEW) {
+open class CreateLambdaFunction() : AnAction(message("lambda.create_new"), null, AwsIcons.Actions.LAMBDA_FUNCTION_NEW) {
     override fun actionPerformed(event: AnActionEvent) {
         val project = event.getRequiredData(LangDataKeys.PROJECT)
         val runtime = event.runtime()
 
-        val dialog = if (handlerName != null) {
-            EditFunctionDialog(project = project, mode = NEW, runtime = runtime, handlerName = handlerName)
-        } else {
-            EditFunctionDialog(project = project, mode = NEW, runtime = runtime)
-        }
+        val dialog = getEditFunctionDialog(project, runtime)
 
         dialog.show()
     }
+
+    protected open fun getEditFunctionDialog(project: Project, runtime: Runtime?): EditFunctionDialog =
+        EditFunctionDialog(project = project, mode = NEW, runtime = runtime)
+}
+
+class ConditionalCreateLambdaFunction(
```

Comment:
I don't understand the benefit of this additional class? I'd much rather just roll this logic into `CreateLambdaFunction`, if a handler is specified, we validate that it's not a member of the template as part of the update, if there's no handler (e.g it was created from the `plugin.xml`) then we skip that logic.

Reply:
I wasn't a fan of the additional null checks. I'm re-combining the classes.

Reply:
`null` is a first-class thing in Kotlin - it's built-in to the language :P